server/commands/write.go

```diff
@@ -87,6 +113,72 @@ func (c *WriteCommand) validateTuplesets(ctx context.Context, req *openfgapb.Wri
 	return nil
 }
 
+// validateTypesForTuple makes sure that when writing a tuple, the types are compatible.
+// 1. If the tuple is of the form (person:bob, reader, doc:budget), then the type "doc", relation "reader" allows type "person".
+// 2. If the tuple is of the form (group:abc#member, reader, doc:budget), then the type "doc", relation "reader" must allow type "group", relation "member".
+// 3. If the tuple is of the form (*, reader, doc:budget), we allow it only if the type "doc" relation "reader" allows at least one type (with no relation)
+func (c *WriteCommand) validateTypesForTuple(authModel *openfgapb.AuthorizationModel, schemaVersion typesystem.SchemaVersion, tk *openfgapb.TupleKey, dbCallsCounter utils.DBCallCounter) error {
```

Comment:
This is a static method. No need for it to be attached to an instance of `*WriteCommand`.

Reply:
I added a `log.warn` in this method, so I need this :(

Reply:
I'm confused.. i don't want to return an error 🤔 I want to continue execution assuming a 1.0 model.